File: system/monolith/java/src/main/java/com/optivem/eshop/monolith/core/dtos/BrowseOrderHistoryResponse.java
package com.optivem.eshop.monolith.core.dtos;

import com.optivem.eshop.monolith.core.entities.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

@Data
public class BrowseOrderHistoryResponse {
    private List<BrowseOrderHistoryItemResponse> orders;

    @Data
    public static class BrowseOrderHistoryItemResponse {
        private String orderNumber;
        private Instant orderTimestamp;
        private String sku;
        private int quantity;
        private BigDecimal totalPrice;
        private OrderStatus status;
    }
}

File: system/monolith/dotnet/Core/Dtos/BrowseOrderHistoryResponse.cs
using Optivem.Shop.Monolith.Core.Entities;

namespace Optivem.Shop.Monolith.Core.Dtos;

public class BrowseOrderHistoryResponse
{
    public List<BrowseOrderHistoryItemResponse> Orders { get; set; } = new();
}

public class BrowseOrderHistoryItemResponse
{
    public string OrderNumber { get; set; } = null!;
    public DateTime OrderTimestamp { get; set; }
    public string Sku { get; set; } = null!;
    public int Quantity { get; set; }
    public decimal TotalPrice { get; set; }
    public OrderStatus Status { get; set; }
}
